doc/classes/CodeHighlighter.xml

```diff
@@ -20,121 +20,278 @@
 				If [param line_only] is [code]true[/code] or [param end_key] is an empty [String], the region does not carry over to the next line.
 			</description>
 		</method>
-		<method name="add_keyword_color">
+		<method name="add_keyword">
+			<return type="void" />
+			<param index="0" name="keyword" type="String" />
+			<param index="1" name="color" type="Color" />
+			<param index="2" name="style" type="int" enum="SyntaxHighlighter.SyntaxFontStyle" default="0" />
+			<description>
+				Sets the color and font style applied to the given keyword.
+				[param keyword] cannot contain any symbols except [code]_[/code].
+			</description>
+		</method>
+		<method name="add_keyword_color" deprecated="Use [method add_keyword] instead.">
 			<return type="void" />
 			<param index="0" name="keyword" type="String" />
 			<param index="1" name="color" type="Color" />
 			<description>
 				Sets the color for a keyword.
-				The keyword cannot contain any symbols except '_'.
+				[param keyword] cannot contain any symbols except [code]_[/code].
 			</description>
 		</method>
-		<method name="add_member_keyword_color">
+		<method name="add_member_keyword">
```

Comment:
I started testing this with my use case which is a markdown highlighter.

This method is too rigid with the arguments. Although not ideal I think the Dictionary should be used here because
- users won't have to guess the order of arguments
- better control over overrides, we can skip color change for example
- adding new styles doesn't overcrowd the function arguments

Additionally I think a new flag needs to be added along the lines of ` allow_recursion` to allow keyword regions to get highlighted in a bold region or easily combining bold and italics.

Example:
- `** * var * **`
Var gets key word highlighted and bold/italics